### PR TITLE
push py-specific stuff down, pull generic stuff up

### PR DESCRIPTION
**Pull request by nicolaskruchten**
@T4rk1n please review :)

### CODE REVIEW

File: dash/development/component_generator.py [hunk 1]
```diff
@@ -10,16 +10,34 @@
...
+            generator(
+                component_name,
+                component_data['props'],
+                component_data['description'],
+                project_shortname
+            )
+
+    return components
+
+def generate_suite_files(project_shortname, components, metadata, *suite_generators):
```
Comment (nicolaskruchten) [inline dash/development/component_generator.py]: I'm pushing the writing of `metadata.json` into this function here, as the path will depend on the language.

**Comment by nicolaskruchten** [comment]
@T4rk1n I can't get the autoformatting rules to work... can you format this as required please and add a commit? :)

**Comment by T4rk1n** [comment]
Look like it worked.

File: dash/development/_py_components_generation.py [hunk 2]
```diff
@@ -164,7 +165,7 @@ def generate_class_file(typename, props, description, namespace):
     print('Generated {}'.format(file_name))
 
 
-def generate_imports(project_shortname, components):
+def generate_imports(project_shortname, components, metadata=None):
```
Comment (T4rk1n) [inline dash/development/_py_components_generation.py]: I think R will need the package.json info too, we should also pass that argument too.

Comment (nicolaskruchten) [reply dash/development/_py_components_generation.py]: Can we add that in at the same time as the R stuff instead?

Comment (T4rk1n) [reply dash/development/_py_components_generation.py]: Actually I think it's just a confusion, `metadata` should be `package_info`. We already mapped the `metadata.json` in generate_component_files, it's the `package.json` mapping that is needed for R. So just rename the argument to `package_info`.

Comment (nicolaskruchten) [reply dash/development/_py_components_generation.py]: That doesn’t make sense to me. The file dump line I moved writes to metadata.json

Comment (nicolaskruchten) [reply dash/development/_py_components_generation.py]: Is it wrong on master now?

Comment (T4rk1n) [reply dash/development/_py_components_generation.py]: No, I forgot about the move of the metadata write, let's keep this as is and we'll modify the functions as needed in the R generation PR.

**Comment by alexcjohnson** [comment]
@T4rk1n looks like there's still some useful stuff here, but might be easier to just recreate it with a new PR? Or would you prefer to ignore this for now and just take care of it in oomph?

**Comment by T4rk1n** [comment]
> Or would you prefer to ignore this for now and just take care of it in oomph?

Ignore for now, this is just a refactor and the underlying code has changed.

**Pull request closed without merging**